contentcuration/contentcuration/utils/nodes.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+from django.conf import settings
+from django.core.files import File as DjFile
+from django.core.files.storage import default_storage
+from django_s3_storage.storage import S3Error
+
+from contentcuration.models import Language, User, ContentNode, FormatPreset, generate_object_storage_name, File
+
+def map_files_to_node(user, node, data):
+    """
+    Generate files that reference the content node.
+    """
+    if settings.DEBUG:
+        # assert that our parameters match expected values
+        assert isinstance(user, User)
+        assert isinstance(node, ContentNode)
+        assert isinstance(data, list)
+
+    # filter for file data that's not empty;
+    valid_data = filter_out_nones(data)
+
+    for file_data in valid_data:
+        filename = file_data["filename"]
+        checksum, ext = filename.split(".")
+
+        # Determine a preset if none is given
+        kind_preset = FormatPreset.get_preset(file_data["preset"]) or FormatPreset.guess_format_preset(filename)
+
+        file_path = generate_object_storage_name(checksum, filename)
+        storage = default_storage
+
+        if not storage.exists(file_path):
+            raise IOError('{} not found'.format(file_path))
+
+        try:
+            if file_data.get('language'):
+                # TODO: Remove DB call per file?
+                file_data['language'] = Language.objects.get(pk=file_data['language'])
+        except ObjectDoesNotExist as e:
+            invalid_lang = file_data.get('language')
+            logging.warning("file_data with language {} does not exist.".format(invalid_lang))
+            return ValidationError("file_data given was invalid; expected string, got {}".format(invalid_lang))
+
+        resource_obj = File(
+            checksum=checksum,
+            contentnode=node,
+            file_format_id=ext,
+            original_filename=file_data.get('original_filename') or 'file',
+            source_url=file_data.get('source_url'),
+            file_size=file_data['size'],
+            preset=kind_preset,
```

Comment:
The real meat of this PR is this single line removal (not shown here). I removed the line saying:
```
file_on_disk=DjFile(storage.open(file_path, "rb"))
```

The line `resource_obj.file_on_disk.name = file_path` down below is enough for Django to identify where the file is on object storage and read from it.